src/lib/containers/FluidModal.tsx

```diff
@@ -0,0 +1,91 @@
+import React from 'react'
+import { Button, Modal } from 'react-bootstrap'
+import { ButtonVariant } from 'react-bootstrap/esm/types'
+import Typography from '../utils/typography/Typography'
+
+type ModalAction = {
+  variant?: ButtonVariant
+  copy: React.ReactNode
+  onClick: () => void
+}
+
+export type FluidModalProps = {
+  show: boolean
+  children: JSX.Element
+  title: string
+  onClose: () => void
+  primaryAction?: ModalAction
+  secondaryActions?: ModalAction[]
+  tertiaryActions?: ModalAction[]
+}
+
+function ModalActionButton(props: Required<ModalAction>) {
+  return (
+    <Button variant={props.variant} onClick={props.onClick}>
+      {props.copy}
+    </Button>
+  )
+}
+
+/**
+ * Full-screen modal that scales with screen size. Fits requirements defined in SWC-5801
+ * @param props
+ * @returns
+ */
+export const FluidModal = (props: FluidModalProps) => {
```

Comment:
This partially addresses [SWC-5801](https://sagebionetworks.jira.com/browse/SWC-5801), which specifies some rules are modals should follow. The bulk of the remaining work for the that issue will just be configuring those rules for the GwtBootstrap3 modal. For new react components, we can use this.